### PR TITLE
rotate the kubeconfig for the controller manager

### PR DESCRIPTION
**Pull request by deads2k**
Instead of running with a kubeconfig file that is baked in during installation, use a kubeconfig based on values that are plumbed from the installer, use a kubeconfig maintained via operator and properly rotated.

/assign @sttts @mfojtik 

I suspect that we will end up having to expose a subset of service-ca in openshift-config-managed to wire through to here and the scheduler.

### CODE REVIEW

File: bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+metadata:
+  name: controller-manager-kubeconfig
+  namespace: openshift-kube-controller-manager
+data:
+  kubeconfig: |
+    apiVersion: v1
+    clusters:
+      - cluster:
+          certificate-authority: /etc/kubernetes/static-pod-resources/configmaps/serviceaccount-ca/ca-bundle.crt
+          server: https://localhost:6443
```
Comment (mfojtik) [inline bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml]: why localhost?

File: bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          cluster: loopback
+          user: kube-controller-manager
+        name: kube-controller-manager
+    current-context: kube-controller-manager
+    kind: Config
+    preferences: {}
+    users:
+      - name: kube-controller-manager
+        user:
+          client-certificate: /etc/kubernetes/static-pod-resources/secrets/kube-controller-manager-client-cert-key/tls.crt
```
Comment (sttts) [inline bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml]: clever

File: bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml (hunk 1 above)
Comment (deads2k) [reply bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml]: > why localhost?

we are on the hostnetwork and we cannot use the service network. it's this or individual master IPs.  this is easier for my head to understand.

File: bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml (hunk 2 above)
Comment (deads2k) [reply bindata/v3.11.0/kube-controller-manager/kubeconfig-cm.yaml]: > clever

yeah, cool feature rigth?

**Comment by deads2k** [comment]
no nay-sayers, tagging

**Comment by deads2k** [comment]
machine api again

/retest

**Comment by deads2k** [comment]
similar to a failure pattern in openshift-apiserver

/retest

**Comment by deads2k** [comment]
"error: unable to parse image registry.svc.ci.openshift.org/ci-op-lpk6ybth/stable@sha256:c21c232054aa2cdc1e34d5bcb7367c959ad01920257985b02225cbc2e6c8df2c: cannot retrieve image configuration for manifest "

/retest

**Comment by mfojtik** [comment]
oh fun, aws flakes after bucket shortage...

@deads2k i would button merge this if you tell me the authorization failures in e2e tests that ran are not related to this PR :-)

**Comment by deads2k** [comment]
> oh fun, aws flakes after bucket shortage...
> 
> @deads2k i would button merge this if you tell me the authorization failures in e2e tests that ran are not related to this PR :-)

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-kube-controller-manager-operator/154/pull-ci-openshift-cluster-kube-controller-manager-operator-master-e2e-aws/637/ from 3:30 yesterday fails on unrelated, known apiserver restart flake.

**Pull request merged**